plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveUpdatablePageSource.java

```diff
@@ -116,28 +117,31 @@ public HiveUpdatablePageSource(
     @Override
     public void deleteRows(Block rowIds)
     {
-        List<Block> blocks = rowIds.getChildren();
-        checkArgument(blocks.size() == 3, "The rowId block for DELETE should have 3 children, but has %s", blocks.size());
         deleteRowsInternal(rowIds);
     }
 
     private void deleteRowsInternal(Block rowIds)
     {
         int positionCount = rowIds.getPositionCount();
-        List<Block> blocks = rowIds.getChildren();
+        ColumnarRow columnarRow = ColumnarRow.toColumnarRow(rowIds);
+        for (int position = 0; position < positionCount; position++) {
+            // Throw an exception if there are any null rows
+            checkArgument(!columnarRow.isNull(position), "In the deleteRows block, found null position %s", position);
```

Comment:
`deleteRows` is not a variable or anything else known

```suggestion
            checkArgument(!columnarRow.isNull(position), "In the delete rowIds, found null row at position %s", position);
```

Reply:
Replaced.